### PR TITLE
Revamp Website

### PR DESCRIPTION
**Pull request by CalebJamesStevens**
# Description:
This pull request involves a comprehensive update to the directory structure and codebase of the Next.js website. The changes primarily involve relocating the content from the 'pages' directory to the 'app' directory, along with a substantial refactor of the codebase and design to align with the new structure. These changes improve organization, maintainability, and overall codebase health.

## Changes
- App directory from Next13 is now in use
- Design was heavily changed while keep the sections relatively the same
- Newsletter subscription was moved to the top of the site
- All other forms were condensed into one contact form

#### Desktop
bendjs com_ Large

#### Mobile
bendjs com_ (1)

#### Desktop
localhost_3002_ (2)

#### Mobile
localhost_3002_ (1)

### CODE REVIEW

**Comment by CalebJamesStevens** [comment]
Sorry about the huge images

I decided to just statically host the member images, not a great reason to integrate a cms into that right now.

I'm going to do another accessibility sweep and then update the forms to have real actions, just looking to see if there's any noticeable code issue.

File: src/app/HeroSection/HeroSection.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <section className='bg-hero-img'>
+      <div className='md:py-32 md:pl-28 md:pr-0 p-10'>
+        <div className='mb-10 md:mb-20'>
+          <Image src='/images/logo_mountains.svg' width={272} height={58.37} alt=''/>
+          <h1 className='text-6xl font-bold'>Bend JS</h1>
+          <p className='text-lg'>A Javascript meetup in Bend, Oregon</p>
+        </div>
+        <p className='text-lg font-bold'>
+          Subscribe to our news letter
```
Comment (chriscoyier) [inline src/app/HeroSection/HeroSection.tsx]:
```suggestion
          Subscribe to our newsletter
```

File: src/app/HeroSection/HeroSection.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        </p>
+        <div>
+          <form className='mt-10 flex items-center md: flex-col md:flex-row gap-2'>
+            <input type='full_name' placeholder='Full Name' className='w-full md:w-fit text-xs rounded p-2' />
+            <input type='email' placeholder='Email Address' className='w-full md:w-fit text-xs rounded p-2'/>
+            <button type='submit' className='w-full md:w-fit rounded bg-black text-white text-xs p-2'>Subscribe</button>
+          </form>
+        </div>
+      </div>
+      {/* <Image src='/images/hero_image.jpg' fill sizes={'(max-width: 768px) 50vw'} alt=''/> */}
```
Comment (chriscoyier) [inline src/app/HeroSection/HeroSection.tsx]:
```suggestion
```

File: src/app/SlackInvite/SlackInvite.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <section className='bg-black text-bendjs-yellow flex flex-col items-center justify-center px-10 p-0 py-16'>
+      <div className='flex flex-col justify-left max-w-lg gap-4'>
+        <div className='flex items-center gap-4'>
+          <Image alt='Slack Logo' src='/images/slack-mark.svg' width={64} height={64} />
+          <h2 className='text-2xl font-bold'>
+            Join Our Slack Community
+          </h2>
+        </div>
+        <p>Join our Slack Community and be the first to hear about new release, ask any questions or request for features!</p>
+        <a className='bg-bendjs-yellow text-black text-xs w-fit px-8 py-5'>Join Now</a>
```
Comment (chriscoyier) [inline src/app/SlackInvite/SlackInvite.tsx]:
```suggestion
<a
          href="https://bendjs.slack.com/ssb/redirect"
          className="w-fit bg-bendjs-yellow px-8 py-5 text-xs text-black"
        >
          Join Now
        </a>
```

**Comment by kcloud99** [comment]
Couple questions @CalebJamesStevens 
1. Why did we remove the CMS integration... I get that it's not robust right now, but does leave us open for more functionality in the future
2. Not sure I'm a fan of the new directory structure with all our components under the same folder as the main pages and css. Makes it messy to try and find anything without being able to just minimize all of the UI components together into a single folder
3. Looks like you're using typescript but didn't add it as a direct dependency to the project. Please add TS and @react/typescript to the dependencies so it will build
4. Love the design updates!

File: src/app/HeroSection/HeroSection.tsx (hunk 1 above)
Comment (CalebJamesStevens) [reply src/app/HeroSection/HeroSection.tsx]: Fixed that

File: src/app/HeroSection/HeroSection.tsx (hunk 2 above)
Comment (CalebJamesStevens) [reply src/app/HeroSection/HeroSection.tsx]: Thanks got that

File: src/app/SlackInvite/SlackInvite.tsx (hunk 3 above)
Comment (CalebJamesStevens) [reply src/app/SlackInvite/SlackInvite.tsx]: Updated

**Comment by CalebJamesStevens** [comment]
> Couple questions @CalebJamesStevens
> 
> 1. Why did we remove the CMS integration... I get that it's not robust right now, but does leave us open for more functionality in the future
> 2. Not sure I'm a fan of the new directory structure with all our components under the same folder as the main pages and css. Makes it messy to try and find anything without being able to just minimize all of the UI components together into a single folder
> 3. Looks like you're using typescript but didn't add it as a direct dependency to the project. Please add TS and @react/typescript to the dependencies so it will build
> 4. Love the design updates!

Hey! Great questions.

The cms was removed because it was a lot of overhead for minimal use, it might make more sense in the future but if you're alright with it holding off on integrating the cms until it has a greater use case might help with the maintenance of the site. This is just the thoughts @chriscoyier and i hashed out. Open to different ideas though.

I've moved the components into a `components` folder. This seems to be the way nextjs handles components in their app router example [https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing](https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing)

I made sure to update the repo to includes types and ts in the dependencies.

And thanks!

Let me know if you have any other concerns

**Comment by chriscoyier** [comment]
You couldn't even spin up the site with the CMS in place without knowing who to ask for the API Keys you needed to put in place via ENV variables and stuff. Fairly big barrier to contribution.

**Comment by CalebJamesStevens** [comment]
> You couldn't even spin up the site with the CMS in place without knowing who to ask for the API Keys you needed to put in place via ENV variables and stuff. Fairly big barrier to contribution.

Agreed

File: public/images/arcadia-floor-plan.svg
Comment (ctsstc) [inline public/images/arcadia-floor-plan.svg:1]: Not sure what's this is doing here?

File: public/.DS_Store
Comment (ctsstc) [inline public/.DS_Store:1]: If we don't have `.DS_Store` added to the gitignore, we should and then remove this from this PR.

**Review by ctsstc — COMMENTED** [review]
I was going to ask where all the tests went, but that's because those components no longer exist.

File: public/images/arcadia-floor-plan.svg
Comment (CalebJamesStevens) [reply public/images/arcadia-floor-plan.svg:1]: Lol saved something personal there on accident nice catch

**Comment by ctsstc** [comment]
@kcloud99 I say we get this integrated and released 🎉 

If there's any additional changes those can be separate pull requests, since there's a lot going on with this larger modernization refactor for NextJS. Also Hacktober is right around the corner as well 🔥 

Some things to bring back as new ticket items:
- Links to sponsors are not linking like they previously were
- Add link back to github for contributing to the site
- Add back social links for youtube and meetup group, along with slack potentially (there is a link, but it should be part of "social" links I think); maybe Github is part of this social link area.
- Button and form consistency - text and stylistic look (rounded vs square corners)

LGTM™ https://deploy-preview-29--friendly-dubinsky-eae4c4.netlify.app/

**Comment by CalebJamesStevens** [comment]
@chriscoyier @kcloud99 

Any objections to merging?

**Comment by CalebJamesStevens** [comment]
@chriscoyier @kcloud99 @ctsstc 
I dont have access to the netfliy account so if there's any other further actions I will need assistance to get this live

**Comment by chriscoyier** [comment]
> @chriscoyier @kcloud99 @ctsstc I dont have access to the netfliy account so if there's any other further actions I will need assistance to get this live

Looks live to me: https://bendjs.com/

I think whatever is on master goes out.

**Pull request merged**